lib/eslint.js

```diff
@@ -648,6 +651,12 @@ module.exports = (function() {
             this.reset();
         }
 
+        if (typeof textOrParseResult === "string") {
+            text = textOrParseResult;
+        } else {
+            text = textOrParseResult.text || "";
```

Comment:
I'm not sure the parse result from espree contains a `text` property, I think it just returns AST and tokens (and a few other things).  I believe this is why @nzakas suggested in https://github.com/eslint/eslint/issues/1025#issuecomment-47809989 to create an object which contains both the text as well as the result from parsing (AST + tokens).